odoo/openupgrade/openupgrade_loading.py

```diff
@@ -0,0 +1,285 @@
+                AND ir_module_module.name in %s
+                AND dep.name not in %s
+            """, (tuple(dependencies), tuple(module_list),))
+
+        dependencies = [x[0] for x in cr.fetchall()]
+        module_list += dependencies
+
+    # Select auto_install modules of which all dependencies
+    # are fulfilled based on the modules we know are to be
+    # installed
+    cr.execute("""
+        SELECT name from ir_module_module WHERE state IN %s
+        """, (('installed', 'to install', 'to upgrade'),))
+    modules = list(set(module_list + [row[0] for row in cr.fetchall()]))
+    cr.execute("""
+        SELECT name from ir_module_module m
+        WHERE auto_install IS TRUE
+            AND state = 'uninstalled'
+            AND NOT EXISTS(
+                SELECT id FROM ir_module_module_dependency d
+                WHERE d.module_id = m.id
+                AND name NOT IN %s)
+         """, (tuple(modules),))
+    auto_modules = [
+        row[0] for row in cr.fetchall()
+        if get_module_path(row[0])
+    ]
+    if auto_modules:
+        logger.info(
+            "Selecting autoinstallable modules %s", ','.join(auto_modules))
+        module_list += auto_modules
+
+    # Set proper state for new dependencies so that any init scripts are run
+    cr.execute("""
+        UPDATE ir_module_module SET state = 'to install'
+        WHERE name IN %s AND name NOT IN %s AND state = 'uninstalled'
+        """, (tuple(module_list), tuple(modules_in)))
+    return module_list
+
+
+def log_model(model, local_registry):
+    """
+    OpenUpgrade: Store the characteristics of the BaseModel and its fields
+    in the local registry, so that we can compare changes with the
+    main registry
+    """
+
+    if not model._name:
+        return
+
+    typemap = {'monetary': 'float'}
+
+    # Deferred import to prevent import loop
+    from odoo import models
+
+    # persistent models only
+    if isinstance(model, models.TransientModel):
+        return
+
+    def isfunction(model, k):
+        if (model._fields[k].compute and
+                not model._fields[k].related and
+                not model._fields[k].company_dependent):
+            return 'function'
+        return ''
+
+    def isproperty(model, k):
+        if model._fields[k].company_dependent:
+            return 'property'
+        return ''
+
+    def isrelated(model, k):
+        if model._fields[k].related:
+            return 'related'
+        return ''
+
+    model_registry = local_registry.setdefault(
+        model._name, {})
+    if model._inherits:
+        model_registry['_inherits'] = {
+            '_inherits': str(model._inherits)}
+    for k, v in model._fields.items():
+        properties = {
+            'type': typemap.get(v.type, v.type),
+            'isfunction': isfunction(model, k),
+            'isproperty': isproperty(model, k),
+            'isrelated': isrelated(model, k),
+            'relation': v.comodel_name if v.type in (
```

Comment:
```suggestion
            'relation': _get_relation(v)
```
call a new method here like:

```python
def _get_relation(self, values):
    if v.type in ('many2many', 'many2one', 'one2many'):
        return v.comodel_name
    elif v.type = 'many2one_reference':
        return v.model_field
    else:
       return  ''
```